### PR TITLE
Add new sorting feature in TutorProMax 

### LINKED ISSUE #42: Add sorting feature to TutorProMax
Add sorting feature to allow users to sort by Name, Phone, Address, Email, TuitionTime.

### PR DESCRIPTION
**Pull request by brvndonkoh**
✨ Summary
This PR introduces the sorting feature to TutorProMax, allowing users to sort the tutor list based on various fields and in different orders (ascending or descending).

🚀 New Functionality
✅ SortCommand
Command word: sort

Users can now sort tutors by the following fields:
- name
- phone
- email
- address
- tuition / tuitionTime

Supports both ascending (asc) and descending (desc) order.

Examples:
`sort name asc`
`sort tuition desc`
`sort by email asc
`
🧠 TuitionTime Sorting Logic
Implemented TuitionTimeUtil.getSortKey(String) to allow comparison of tuition time values.
Converts a value like "Tuesday, 1800-2000" into a sortable key "2_1800", enabling chronological ordering based on day and start time.

🧪 Tests Added
SortCommandTest: Validates sorting logic, exception handling, and equality checks.
SortCommandParserTest: Covers parsing for various valid and invalid input formats (e.g., spacing, missing arguments, invalid directions).
TuitionTimeUtilTest: Tests for correct sort key generation, edge cases (e.g., invalid/unknown day formats).

🛠️ Other Improvements
Introduced a ModelStub for unit testing SortCommand in isolation.
Resolved Checkstyle violations (indentation, ordering, import style, etc.)
Ensured compatibility with existing model and command architecture.

Fixes #42

### CODE REVIEW

**Review by davidchongg — APPROVED** [review]
Good work

File: src/main/java/seedu/address/logic/commands/DeleteCommand.java [hunk 1]
```diff
@@ -31,6 +31,13 @@ public DeleteCommand(Index targetIndex) {
         this.targetIndex = targetIndex;
     }
 
+    /**
+     * Executes the delete command, which deletes a person from the address book.
+     *
+     * @param model The model which the command should operate on.
+     * @return A {@code CommandResult} indicating the outcome of the delete operation.
+     * @throws CommandException If the provided index is out of bounds (i.e., not a valid person in the current list).
+     */
```
Comment (huangtian666) [inline src/main/java/seedu/address/logic/commands/DeleteCommand.java:40]: Nice javadocs written!!! like how you have such a detailed description!

File: src/main/java/seedu/address/logic/commands/SortCommand.java [hunk 2]
```diff
@@ -0,0 +1,100 @@
...
+            break;
+        case "phone":
+            comparator = Comparator.comparing(p -> p.getPhone().value);
+            break;
+        case "email":
+            comparator = Comparator.comparing(p -> p.getEmail().value.toLowerCase());
+            break;
+        case "address":
+            comparator = Comparator.comparing(p -> p.getAddress().value.toLowerCase());
+            break;
```
Comment (huangtian666) [inline src/main/java/seedu/address/logic/commands/SortCommand.java:64]: Nice switch cases used! This helps to improve the readability of your code!

File: src/main/java/seedu/address/logic/parser/SortCommandParser.java [hunk 3]
```diff
@@ -0,0 +1,45 @@
+package seedu.address.logic.parser;
+
+import seedu.address.logic.commands.SortCommand;
+import seedu.address.logic.parser.exceptions.ParseException;
+
+/**
+    * Parses input arguments and creates a new SortCommand.
+    */
+public class SortCommandParser implements Parser<SortCommand> {
```
Comment (huangtian666) [inline src/main/java/seedu/address/logic/parser/SortCommandParser.java:9]: Good job in creating a new command! Nice OOP!

File: src/test/java/seedu/address/logic/commands/SortCommandTest.java [hunk 4]
```diff
@@ -0,0 +1,169 @@
...
+import org.junit.jupiter.api.Test;
+
+import javafx.collections.FXCollections;
+import javafx.collections.ObservableList;
+import seedu.address.logic.commands.exceptions.CommandException;
+import seedu.address.model.Model;
+import seedu.address.model.person.Person;
+import seedu.address.testutil.PersonBuilder;
+
+public class SortCommandTest {
```
Comment (huangtian666) [inline src/test/java/seedu/address/logic/commands/SortCommandTest.java:21]: WOW so many tests added!

File: src/test/java/seedu/address/model/person/TuitionTimeUtilTest.java [hunk 5]
```diff
@@ -0,0 +1,40 @@
...
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+
+import org.junit.jupiter.api.Test;
+
+public class TuitionTimeUtilTest {
+
+    @Test
+    public void getSortKey_validInput_returnsCorrectSortKey() {
+        assertEquals("1_0900", TuitionTimeUtil.getSortKey("Monday 0900-1100"));
```
Comment (huangtian666) [inline src/test/java/seedu/address/model/person/TuitionTimeUtilTest.java:11]: Nice tests to improve codecov!

**Review by huangtian666 — COMMENTED** [review]
Good job! LGTM.
